services/api.py

```diff
@@ -747,19 +755,18 @@ def service_nodes_by_ancestors(service_node_ids):
 
         service_node_ids = None
         if service_nodes:
-            service_nodes = service_nodes.lower()
-            service_node_ids = service_nodes.split(',')
+            service_node_ids = service_nodes
         elif level_specs:
             if level_specs['type'] == 'include':
                 service_node_ids = level_specs['service_nodes']
         if service_node_ids:
             queryset = queryset.filter(service_nodes__in=service_nodes_by_ancestors(service_node_ids)).distinct()
 
+        service_nodes_excl = filters.get('exclude_service_nodes', None)
+
         service_node_ids = None
-        val = filters.get('exclude_service_nodes', None)
-        if val:
-            val = val.lower()
-            service_node_ids = val.split(',')
+        if service_nodes_excl:
+            service_node_ids = service_nodes_excl
```

Comment:
this will change the logic of filter from:

`
service_nodes = filters.get(...) 
if service_nodes: 
    service_node_ids = service_nodes
elif level_specs: 
    if level_specs['type'] == 'include':
        service_node_ids = something
if service_node_ids: do something with service_node_ids
`

to:
`
service_nodes = filters.get(...)
if level_specs: 
    if level_specs['type'] == 'include':
        service_nodes = something
if service_nodes: do something with service_nodes
`

Is it ok that if query parameters have both: service_node and level with type include, then service_node ids will be overwritten?

Reply:
Yeah, good idea to add the `not service_nodes` condition.